EventSubscriber.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Plugin\ProductRank;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Eccube\Event\EccubeEvents;
+use Eccube\Event\EventArgs;
+use Plugin\ProductRank\Event\Event;
+use Knp\Component\Pager\Event\ItemsEvent;
+
+class EventSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var Event
+     */
+    protected $event;
+
+    /**
+     * EventSubscriber constructor.
+     *
+     * @param Event $event
+     */
+    public function __construct(
+        Event $event
+    ) {
+        $this->event = $event;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            EccubeEvents::FRONT_PRODUCT_INDEX_SEARCH => ['onFrontProductIndexSearch', 10],
```

Comment:
You can use QueryKey::PRODUCT_SEARCH event for these two events

Reply:
@lqdung-lockon @hoand-vn @toannguyen-lockon 
I replace EccubeEvents::FRONT_PRODUCT_INDEX_SEARCH event by using QueryCustomizer. But query customizer can not set option `wrap-queries` of paginator (see https://github.com/KnpLabs/knp-components/blob/master/doc/pager/config.md).

So I can not replace knp_pager.items by QueryCustomizer. Do you have any idea ?